docs/installation/server-config/security.rst

```diff
@@ -1,4 +1,227 @@
+    - ``DESCRIBE`` permission on the ``__consumer_offsets`` topic.
+
+Source topics
+    KSQL users require ``DESCRIBE`` and ``READ`` permissions for each source and input topic. The topic should already exist
+    when KSQL is started.
+
+Sink topics
+    KSQL users require ``DESCRIBE`` and ``WRITE`` permissions ror each sink and output topic.
+
+    By default, KSQL will attempt to create any output topics that do not exist. To allow this the KSQL user will require
+    ``CREATE`` permissions on the ``CLUSTER``. Alternatively, topics can be created manual before running KSQL. To determine
+    the list of output topics their required configuration, for example, partition count, replication factor,
+    retention policy, etc, you can run KSQL on open cluster first.
+
+    If you want KSQL to create sink topics, then the KSQL user should be granted ``CREATE`` permission on the ``CLUSTER``. KSQL
+    will create sink topics whenever you are doing a persistent CTAS/CTAS query.
+
+Change-log and repartition topics
+    Internally KSQL uses repartition topics and changelog topics for selected operations. KSQL users minimally require a
+    ``DESCRIBE``, ``READ``, and ``WRITE`` permissions for each changelog and repartition ``TOPIC``.
+
+    By default, KSQL will attempt to create any repartition or changelog topics that do not exist. To allow this, the KSQL
+    user requires ``CREATE`` permissions on the ``CLUSTER``. Alternatively, topics can be created manually before running KSQL.
+    To determine the list of topics and their required configuration, for example, partition count, replication factor,
+    retention policy, etc, you can run KSQL on open cluster first.
+
+    KSQL requires repartition topics when using either ``PARTITION BY``, or using ``GROUP BY`` on none-key values. KSQL
+    requires changelog topics for any ``CREATE TABLE x AS`` statements.
+
+    All changelog and repartition topics are prefixed with ``_confluent-ksql-<value of ksql.service.id property>_query_<query id>_``
+    where ``ksql.service.id`` defaults to ``ksql_``.
+
+Consumer groups
+    The set of consumer groups that KSQL requires depends on the queries that are being executed. The easiest way to
+    determine the list of consumer groups is to first run the queries on an unsecured or open Kafka cluster and list the
+    groups created. For more information about how to list groups, see
+    `Managing Consumer Groups <http://kafka.apache.org/documentation.html#basic_ops_consumer_group>`__.
+
+    Consumer group names are formatted like ``<value of ksql.service.id property>_query_<query id>``, where the default
+    of ``ksql.service.id`` is ``ksql_``.
+
+    KSQL users require a minimum of ``DESCRIBE`` and ``READ`` permissions for ``GROUP``.
+
+.. _security-config-ref:
+
+Security Configuration Reference
+--------------------------------
+
+------------------------------------------
+Global SSL and SASL configuration for KSQL
```

Comment:
I'm not certain what this section relates to... I it think it can be dropped.

Reply:
deleted lines 172-189